Assets/Mirage/Runtime/InterestManagement/InterestManager.cs

```diff
@@ -83,19 +77,22 @@ private void OnAuthenticated(INetworkPlayer player)
 
             foreach (ObserverData observer in _visibilitySystems)
             {
-                if (player.Identity != null && observer.Observers.ContainsKey(player.Identity))
-                    found = true;
+                foreach (KeyValuePair<NetworkIdentity, HashSet<INetworkPlayer>> variable in observer.Observers)
```

Comment:
Instead of a foreach, use Any() (it will also stop once found)
```suggestion
                    found = observer.Observers.Any(x => x.Value.Contains(player));
```